carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/domain/StudyStatus.kt

```diff
@@ -9,22 +9,53 @@ import kotlinx.serialization.Serializable
  * Describes the status of a [Study]: the number of participants, progress towards study goal, etc.
  */
 @Serializable
-data class StudyStatus(
-    val studyId: UUID,
+sealed class StudyStatus
+{
+    abstract val studyId: UUID
     /**
      * A descriptive name for the study, as assigned by the [StudyOwner].
      */
-    val name: String,
+    abstract val name: String
     /**
      * The date when this study was created.
      */
-    val creationDate: DateTime,
+    abstract val creationDate: DateTime
     /**
      * Determines whether the study in its current state is ready to be deployed to participants.
      */
-    val canDeployToParticipants: Boolean,
+    abstract val canDeployToParticipants: Boolean
     /**
-     * Determines whether a study protocol has been locked in and the study may be deployed to real participants.
+     * Determines whether a study protocol can be set/changed for the study.
      */
-    val isLive: Boolean
-)
+    abstract val canSetStudyProtocol: Boolean
+
+
+    /**
+     * Study status for when a study is being configured.
+     */
+    @Serializable
+    data class Configuring(
+        override val studyId: UUID,
+        override val name: String,
+        override val creationDate: DateTime,
+        override val canDeployToParticipants: Boolean,
+        override val canSetStudyProtocol: Boolean,
+        /**
+         * Determines whether a study is fully configured and can 'go live'.
+         */
+        val canGoLive: Boolean
+    ) : StudyStatus()
+
+
+    /**
+     * Study status for when a study is 'live'.
+     */
+    @Serializable
+    data class Live(
+        override val studyId: UUID,
+        override val name: String,
+        override val creationDate: DateTime,
+        override val canDeployToParticipants: Boolean,
+        override val canSetStudyProtocol: Boolean
+    ) : StudyStatus()
+}
```

Comment:
I think this was actually more or less what I wanted :) I'm still no fan of top-level polymorphic serialization as i think it imposes a very stack specific interpretation of messages passed between systems.  But let's try this for now